src/librustc_trans/back/link.rs

```diff
@@ -141,17 +141,31 @@ pub fn build_link_meta(incremental_hashes_map: &IncrementalHashesMap) -> LinkMet
     return r;
 }
 
-// The third parameter is for an env vars, used to set up the path for MSVC
-// to find its DLLs
+// The third parameter is for env vars, used on windows to set up the
+// path for MSVC to find its DLLs, and gcc to find its bundled
+// toolchain
 pub fn get_linker(sess: &Session) -> (String, Command, Vec<(OsString, OsString)>) {
+    let envs = |linker: Option<&str>| -> Vec<(OsString, OsString)> {
+        if sess.target.target.options.is_like_windows {
+            // On windows-gnu (msvc is handled separately), we bundle
+            // gcc and other tools with the official releases. This is
+            // how the other tools that gcc itself needs are located.
+            vec![("PATH".into(), command_path(sess, linker.map(PathBuf::from)))]
```

Comment:
This doesn't seem right. The second parameter to `command_path` is added to the end of `%Path%` and before #42225 it was only used by MSVC.